lib/computeStatistics.js

```diff
@@ -0,0 +1,182 @@
+
+    count += getNumberOfExternalProperties(gltf.buffers);
+    count += getNumberOfExternalProperties(gltf.images);
+    count += getNumberOfExternalProperties(gltf.shaders);
+
+    return count;
+}
+
+function getNumberOfProperties(object) {
+    return Object.keys(object).length;
+}
+
+function getDrawCallStatistics(gltf) {
+    var numberOfDrawCalls = 0;
+    var numberOfRenderedPrimitives = 0;
+
+    var primitives = PrimitiveHelpers.getAllPrimitives(gltf);
+    var primitivesLength = primitives.length;
+
+    numberOfDrawCalls += primitivesLength;
+
+    for (var m = 0; m < primitivesLength; ++m) {
+        var primitive = primitives[m];
+        var indices = primitive.indices;
+        var indicesCount = gltf.accessors[indices].count;
+
+        switch(primitive.primitive) {
+            case 0: // POINTS
+                numberOfRenderedPrimitives += indicesCount;
+                break;
+            case 1: // LINES
+                numberOfRenderedPrimitives += (indicesCount / 2);
+                break;
+            case 2: // LINE_LOOP
+                numberOfRenderedPrimitives += indicesCount;
+                break;
+            case 3: // LINE_STRIP
+                numberOfRenderedPrimitives += Math.max(indicesCount - 1, 0);
+                break;
+            case 4: // TRIANGLES
+                numberOfRenderedPrimitives += (indicesCount / 3);
+                break;
+            case 5: // TRIANGLE_STRIP
+            case 6: // TRIANGLE_FAN
+                numberOfRenderedPrimitives += Math.max(indicesCount - 2, 0);
+                break;
+        }
+    }
+
+    return {
+        numberOfDrawCalls : numberOfDrawCalls,
+        numberOfRenderedPrimitives : numberOfRenderedPrimitives
+    };
+}
+
+/**
+ * Returns an object containing the statistics for the glTF assest.
+ *
+ * @param {Object} gltf A javascript object containing a glTF asset with extras and defaults.
+ * @returns {Statistics} Object containing the properties of the assest.
+ *
+ * @see Statistics
+ * @see printStatistics
+ */
+function getStatistics(gltf) {
+    if (!defined(gltf)) {
+        return new Error('gltf must be defined');
+    }
+
+    var drawStats = getDrawCallStatistics(gltf);
+
+    var statistics = new Statistics();
+    statistics.buffersSizeInBytes          = getBuffersSize(gltf.buffers);
+    statistics.numberOfImages              = getNumberOfProperties(gltf.images);
+    statistics.numberOfExternalRequests    = getNumberOfExternalRequests(gltf);
+    statistics.numberOfDrawCalls           = drawStats.numberOfDrawCalls;
+    statistics.numberOfRenderedPrimitives  = drawStats.numberOfRenderedPrimitives;
+    statistics.numberOfNodes               = getNumberOfProperties(gltf.nodes);
+    statistics.numberOfMeshes              = getNumberOfProperties(gltf.meshes);
+    statistics.numberOfMaterials           = getNumberOfProperties(gltf.materials);
+    statistics.numberOfAnimations          = getNumberOfProperties(gltf.animations);
```

Comment:
Usually we don't format like this, not that its wrong but just not consistent with the style generally. Use formatting like:

```
statistics.buffersSizeInBytes = getBuffersSize(gltf.buffers);
statistics.numberOfImages = getNumberOfProperties(gltf.images);
```